wireapp/wire-android
# refactor: Remove NameUpdater and change the way we choose names for conversations

fixes https://wearezeta.atlassian.net/browse/AN-6873

This is ground work for better handling of removed users after the Large Teams feature. We have three main types of conversations: 1-to-1, fake 1-to-1, and group. Group conversations should have names of their own, but some older ones don't. For those which don't, and for all other types, we should use names generated from the list of members.
Until now, we used a field `ConversationData.generatedName` to store the generated name. The name had to be updated every time one of the conversation members changed their name. A special service, `NameUpdater`, did that in response to every name changing event. The name was also updated for real new group conversations which have their own names, so generated names were never used. And on top of that we had messy code in several places in the app which sometimes decided the conversation name should be changed to "Default" when it really shouldn't.

This PR removes `NameUpdater` and makes `generatedName` obsolete. Instead, we will have one place, in `ConversationsService` where we create conversation names on the fly. We will use "Default" only when there is no way to generate the name.

Tests:
1. Create a number of 1-to-1, fake 1-to-1, and group conversations.
2. Change names of members in the conversations:
2a. The names of the real group conversations should stay the same.
2b. The names of 1-to-1 and fake 1-to-1 conversations should change.
3. Remove users from teams.
3a. Conversation names should stay the same although user names can change to "Default" (this will be fixed in another PR)
4. If you have a big conversations list, test if scrolling didn't slow down (in comparison to `develop`)

PS. There was also a minor bug in `CallController` and `CallStartController` where the members of the call couldn't be accessed if the conversation belonged to the other account than the current one. Fixed.

#### APK
[Download build #2088](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2088/artifact/build/artifact/wire-dev-PR2840-2088.apk)
[Download build #2102](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2102/artifact/build/artifact/wire-dev-PR2840-2102.apk)
[Download build #2106](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2106/artifact/build/artifact/wire-dev-PR2840-2106.apk)
